Gems/Atom/Feature/Common/Code/Source/CoreLights/ProjectedShadowmapsPass.cpp

```diff
@@ -60,302 +46,64 @@ namespace AZ
-        {
-            if (AZStd::find(
-                m_childrenPipelineViewTags.begin(),
-                m_childrenPipelineViewTags.end(),
-                viewTag) != m_childrenPipelineViewTags.end() &&
-                outPassesByDrawList.find(m_drawListTag) == outPassesByDrawList.end())
-            {
-                outPassesByDrawList[m_drawListTag] = this;
-                outDrawListMask.set(m_drawListTag.GetIndex());
-            }
-        }
-
-        RPI::Ptr<ShadowmapPass> ProjectedShadowmapsPass::CreateChild(size_t childIndex)
-        {
-            const Name passName{ AZStd::string::format("ProjectedShadowmapPass.%zu", childIndex) };
-
-            auto passData = AZStd::make_shared<RPI::RasterPassData>();
-            passData->m_drawListTag = m_drawListTagName;
-            passData->m_pipelineViewTag = GetPipelineViewTagOfChild(childIndex);
-
-            return ShadowmapPass::CreateWithPassRequest(passName, passData);
-        }
-
-        void ProjectedShadowmapsPass::CreateClearShadowDrawPacket()
-        {
-            // Force load of shader to clear shadow maps.
-            const AZStd::string clearShadowShaderFilePath = "Shaders/Shadow/ClearShadow.azshader";
-            Data::Asset<RPI::ShaderAsset> shaderAsset = RPI::AssetUtils::LoadCriticalAsset<RPI::ShaderAsset>
-                (clearShadowShaderFilePath, RPI::AssetUtils::TraceLevel::Assert);
-
-            m_clearShadowShader = AZ::RPI::Shader::FindOrCreate(shaderAsset);
-            const AZ::RPI::ShaderVariant& variant = m_clearShadowShader->GetRootVariant();
-
-            AZ::RHI::PipelineStateDescriptorForDraw pipelineStateDescriptor;
-            variant.ConfigurePipelineState(pipelineStateDescriptor);
-
-            GetScene()->ConfigurePipelineState(m_clearShadowShader->GetDrawListTag(), pipelineStateDescriptor);
-
-            AZ::RHI::InputStreamLayoutBuilder layoutBuilder;
-            pipelineStateDescriptor.m_inputStreamLayout = layoutBuilder.End();
-
-            const AZ::RHI::PipelineState* pipelineState = m_clearShadowShader->AcquirePipelineState(pipelineStateDescriptor);
-            if (!pipelineState)
-            {
-                AZ_Assert(false, "Shader '%s'. Failed to acquire default pipeline state", shaderAsset->GetName().GetCStr());
-                return;
-            }
-
-            AZ::RHI::DrawPacketBuilder drawPacketBuilder;
-            drawPacketBuilder.Begin(nullptr);
-            drawPacketBuilder.SetDrawArguments(AZ::RHI::DrawLinear(1, 0, 3, 0));
-
-            AZ::RHI::DrawPacketBuilder::DrawRequest drawRequest;
-            drawRequest.m_listTag = m_clearShadowShader->GetDrawListTag();
-            drawRequest.m_pipelineState = pipelineState;
-            drawRequest.m_sortKey = AZStd::numeric_limits<RHI::DrawItemSortKey>::min();
-
-            drawPacketBuilder.AddDrawItem(drawRequest);
-            m_clearShadowDrawPacket = drawPacketBuilder.End();
+            QueueForBuildAndInitialization();
```

Comment:
Can you add comment (in the code) about why this QueueForBuildAndInitialization is called here?